Add some more statistics, Modify game ticks

RedSparr0w opened the pull request:
Add new statistics:
- Daily Deal Trades
- Total Berries Harvested
- Quests Completed
- Total Shards Gained

Fix game tick speed:
Should be checking if the current tick matches or above instead of just above, otherwise the tick doesn't happen until 1.1 seconds instead of 1 second.

RedSparr0w committed:
Add Daily Deal Trades statistic

RedSparr0w committed:
Add total berries harvested statistic

RedSparr0w committed:
Add Quests Complete statistic

RedSparr0w committed:
totalShards -> shardsGained, Add total shards gained

RedSparr0w committed:
Update Changelog.ts

RedSparr0w committed:
Fixup version number

RedSparr0w committed:
Update GainShardsQuest.ts

RedSparr0w committed:
Add statistic Seconds Played

RedSparr0w committed:
Move the tick counter incrementals

RedSparr0w committed:
Fix game tick counters

Aegyo commented on src/scripts/Game.ts:
```diff
@@ -179,14 +181,19 @@ class Game {
-            GameHelper.updateTime();
+        // Game timers
+        GameHelper.counter += GameConstants.TICK_TIME;
+        if (GameHelper.counter >= GameConstants.SECOND) {
```
Should this be `GameConstants.MINUTE`?

RedSparr0w replied:
```suggestion
        if (GameHelper.counter >= GameConstants.MINUTE) {
```

Aegyo reviewed (commented):
:+1:
The version check may need an update after develop is merged up into this.
Also a potential addition to this - routeKills needs to be split into region and route before we add Unova, since those routes start back at 1. May prefer to do that in a separate PR though.

RedSparr0w committed:
SECOND -> MINUTE

RedSparr0w committed:
Merge branch 'develop' into more-statistics

RedSparr0w committed:
Add hatched/shiny hatched statistic

RedSparr0w committed:
Update 0.5.2 update error message on fail

RedSparr0w commented:
> Also a potential addition to this - routeKills needs to be split into region and route before we add Unova, since those routes start back at 1. May prefer to do that in a separate PR though.

Will do that in a different PR, not entirely sure how we will go about it.
As it won't work with any of the current implementations,
Unless we use the object and merge the region and route number into 1 (`region_number|route_number`),
But that feels a little bit messy.

RedSparr0w commented:
I've also added 2 more statistics since the last review:
- Pokemon Hatched
- Shiny Pokemon Hatched

I've added to the update function to pull our current method of hatched statistics into the new one.

RedSparr0w committed:
Increase damage output when you hatch a pokemon of the same type if you have already obtained that pokemon
